src/transform.php

```diff
@@ -47,22 +53,12 @@ function callMethod($methodName, ...$args)
  *                              into the array.
  *
  * @return callable
+ *
+ * @deprecated
```

Comment:
Do you not think that having a `getElement` for array access and `getProperty` for object property access is a good idea? There is a possibility that you have an object which has public properties and implements `ArrayAccess`.